Automattic/jetpack
# Launchpad: migrate business logic to Task definitions

See Automattic/wp-calypso#75504

## Proposed changes:
This finishes work begun in #30397 to consolidate all task-completing logic from the old mu-plugin to the Task registry, using `add_listener_callback`.

### Other information:

- [x] Have you written new tests for your changes, if applicable? **Fixed tests **

## Jetpack product discussion
n/a

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:

1. apply this to your wpcom sandbox
2. sandbox `public-api.wordpress.com`
3. sandbox the URL of the site **you create before you create it** (eg `wiebelaunchpadwrite.wordpress.com`)

With each type of Launchpad-supporting site, go through the full flow and ensure that:

1. All tasks are marked off when they should be
2. Launching works

## Known issues
The VideoPress upload detector does not work properly. I've added a workaround in 284c921e4726976fec70ddd12281e69d5376652e that mimics the old mu-plugin logic that worked. Unfortunately this can't be tested until this lands in production because it needs to be on job servers.
